cookstyle/ConditionalBooleanLiteralCleanup.rb

```diff
@@ -0,0 +1,226 @@
+# Copyright (c) 2025-present, Meta Platforms, Inc. and affiliates
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+require 'rubocop'
+
+# TODO(dcrosby) This will eventually be upstreamed as a fix to the
+# boolean literal cops in RuboCop, but for now this addresses some
+# cleanup that other Cookstyle cops do (which is why this is a Cookstyle
+# cop, and not in our RuboCop-only config).
+
+# A caveat with this linter is that it cannot determine `if false`
+# because some deprecated thing was converted to `false`, versus you
+# wanting to disable something temporarily with `if false`. It's like in
+# addition to nil and false we need some other way of saying "false,
+# because I'll forget about this reverted diff in 2 months" ;-)
+# In the event you really want to keep that boolean literal condition
+# add a comment with `rubocop:disable Chef/Meta/ConditionalBooleanLiteralCleanup`
```

Comment:
Yes, this is the correct policy answer: if you want to specifically instruct the linter to ignore your seemingly illogical statement, disable the linter for that line.